kratos/includes/model_part.h

```diff
@@ -1457,193 +1498,140 @@ class KRATOS_API(KRATOS_CORE) ModelPart final
-    template<class TContainer, std::enable_if_t<IsRValueContainer<TContainer>::value, bool> = true>
-    void AddGeometries(TContainer&& rInputContainer)
-    {
-        KRATOS_TRY
-
-        // This check is required in cases where std::vector<IndexType> is passed as an r value, then it will
-        // not call the correct addition with the input args of const std::vector<IndexType>&, it will get in to this method.
-        // this check will redirect it to the std::vector<IndexType> method with the rContainer.
-        if constexpr(!std::is_same_v<BaseType<TContainer>, std::vector<IndexType>>) {
-            ModelPart* p_root_model_part = &this->GetRootModelPart();
-
-            block_for_each(rInputContainer.begin(), rInputContainer.end(), [p_root_model_part](const auto& prGeometry) {
-                const auto& r_geometry = ReferenceGetter<GeometryContainerType::value_type>::Get(prGeometry);
-                const auto& r_geometries = p_root_model_part->Geometries();
-                auto it_found = r_geometries.find(r_geometry.Id());
-                if (it_found != p_root_model_part->GeometriesEnd()) {
-                    if (GeometryType::HasSameGeometryType(r_geometry, *it_found)) { // Check the geometry type and connectivities
-                        for (IndexType i_pt = 0; i_pt < r_geometry.PointsNumber(); ++i_pt) {
-                            KRATOS_ERROR_IF((r_geometry)[i_pt].Id() != (*it_found)[i_pt].Id()) << "Attempting to add a new geometry with Id: " << r_geometry.Id() << ". A same type geometry with same Id but different connectivities already exists." << std::endl;
-                        }
-                    } else if(&(*it_found) != &r_geometry) { // Check if the pointee coincides
-                        KRATOS_ERROR << "Attempting to add a new geometry with Id: " << it_found->Id() << ". A different geometry with the same Id already exists." << std::endl;
-                    }
-                }
-            });
-
-            InsertEntities([](ModelPart* pModelPart) {
-                return &(pModelPart->Geometries());
-            }, std::move(rInputContainer));
-        } else {
-            // call the vector of int insertion.
-            AddGeometries(rInputContainer);
-        }
-
-
-
-        KRATOS_CATCH("")
+    /** Returns if the Element corresponding to it's name exists */
+    bool HasGeometry(std::string GeometryName, IndexType ThisIndex = 0) const {
+        return GetMesh(ThisIndex).HasGeometry(GeometryName);
     }
 
     /// Returns the Geometry::Pointer corresponding to the Id
-    typename GeometryType::Pointer pGetGeometry(IndexType GeometryId) {
-        auto i = mGeometries.find(GeometryId);
-        KRATOS_ERROR_IF(i == mGeometries.end()) << " geometry index not found: " << GeometryId << ".";
-        return *(i.base());
+    typename GeometryType::Pointer pGetGeometry(IndexType GeometryId, IndexType ThisIndex = 0) {
+        return GetMesh(ThisIndex).pGetGeometry(GeometryId);
```

Comment:
```suggestion
    typename GeometryType::Pointer pGetGeometry(IndexType GeometryId)
    {
        return GetMesh(0).pGetGeometry(GeometryId);
```
Same in all the other ones.